internal/store/fake_store.go

```diff
@@ -227,3 +232,93 @@ func NewFakeStore(
+		reflect.TypeOf(&configurationv1.KongPlugin{}):                   configurationv1.SchemeGroupVersion.WithKind("KongPlugin"),
+		reflect.TypeOf(&configurationv1.KongClusterPlugin{}):            configurationv1.SchemeGroupVersion.WithKind("KongClusterPlugin"),
+		reflect.TypeOf(&configurationv1.KongIngress{}):                  configurationv1.SchemeGroupVersion.WithKind("KongIngress"),
+		reflect.TypeOf(&configurationv1.KongConsumer{}):                 configurationv1.SchemeGroupVersion.WithKind("KongConsumer"),
+	}
+
+	out := &bytes.Buffer{}
+
+	// fillGVKAndAppendToBuffer is a helper function that sets the GVK of the given object and appends it to the output.
+	fillGVKAndAppendToBuffer := func(obj runtime.Object) error {
+		gvk, ok := typeToGVK[reflect.TypeOf(obj)]
+		if !ok {
+			return fmt.Errorf("unknown type: %T", obj)
+		}
+		obj.GetObjectKind().SetGroupVersionKind(gvk)
+		b, err := dumpObjAsYAML(obj)
+		if err != nil {
+			return err
+		}
+		out.WriteString("---\n")
+		out.Write(b)
+		return nil
+	}
+
+	// Let's gather all objects in a single generic slice.
+	var allObjects []any
+	allObjects = append(allObjects, lo.ToAnySlice(objects.IngressesV1)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.IngressClassesV1)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.HTTPRoutes)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.UDPRoutes)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.TCPRoutes)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.TLSRoutes)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.GRPCRoutes)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.ReferenceGrants)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.Gateways)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.TCPIngresses)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.UDPIngresses)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.IngressClassParametersV1alpha1)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.Services)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.EndpointSlices)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.Secrets)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.KongPlugins)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.KongClusterPlugins)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.KongIngresses)...)
+	allObjects = append(allObjects, lo.ToAnySlice(objects.KongConsumers)...)
+
+	for _, obj := range allObjects {
+		if err := fillGVKAndAppendToBuffer(obj.(runtime.Object)); err != nil {
+			return nil, err
+		}
+	}
+
+	return out.Bytes(), nil
+}
+
+// dumpObjAsYAML dumps the given object as YAML.
+// It uses the JSON printer to dump the object as JSON, and then converts the JSON to YAML because some of the objects
+// we want to dump do not have YAML tags.
+func dumpObjAsYAML(obj runtime.Object) ([]byte, error) {
```

Comment:
To me, dump means that I'd expect something to be printed. How about we name this `objMarshalToYAML`?